src/Tooltip/TooltipTouch.tsx

```diff
@@ -80,40 +94,42 @@ export default class TooltipTouch extends React.PureComponent {
     });
   }
 
-  getLocals({ popover, ...props }) {
+  render() {
+    const { popover: _popover, children, className, id, style, type, size } = this.props as TooltipTouchDefaultedProps;
     const { isOpen } = this.state;
-
-    return {
-      ...props,
+    const popover = Object.assign({
+      ..._popover,
       isOpen,
-      touchListeners: {
-        onTouchStart: this.onTouchStart,
-        onTouchEnd: this.onTouchEnd,
-        onTouchMove: isOpen ? this.onTouchMove : undefined
-      },
-      popover: {
-        ...popover,
-        isOpen,
-        className: cx(popover.className, 'tooltip-touch')
-      },
-      spanStyle: isOpen ? {
-        display: 'hidden',
-        position: 'fixed',
-        top: -1000
-      } : undefined
+      className: cx(_popover.className, 'tooltip-touch')
+    });
+    const spanStyle: React.CSSProperties = {
+      display: 'hidden',
+      position: 'fixed',
+      top: -1000
     };
-  }
 
-  template({ children, className, isOpen, touchListeners, spanStyle, ...locals }) {
     return (
       <View
         vAlignContent='center'
         className={className}
         ref={r => { if (r) { this.ref = ReactDOM.findDOMNode(r); } }}
-        {...touchListeners}
+        onTouchStart={this.onTouchStart}
+        onTouchEnd={this.onTouchEnd}
+        onTouchMove={isOpen ? this.onTouchMove : undefined}
       >
         <span style={spanStyle}>{children}</span>
-        {isOpen && <Tooltip {...locals}>{children}</Tooltip>}
+        {isOpen && (
+          <Tooltip
+            id={id}
```

Comment:
mmm... I'm struggling to follow what's happening here after the removal of `@skinnable`, but it seems we were using a rest operator before and now we're not. Are we sure we're passing all the props? Isn't better to keep the "rest" logic we had before?

Reply:
I double checked and I found out that `className` wasn't there. Will remove. Nonetheless I think it's much more readable without spread operators and future refactors proof

Reply:
I don't know... Here `TooltipTouch` is using the same props defined for `Tooltip` (is not even defining a new ts type in this file), but doing this you're in some way listing the props `TooltipTouch` can receive.
What if you add a new optional prop to `Tooltip` and you completely forget about `TooltipTouch`? That prop will be ignored and you won't even notice it.